winpr/libwinpr/sspi/Kerberos/kerberos.c

```diff
@@ -1594,19 +1520,15 @@ static SECURITY_STATUS SEC_ENTRY kerberos_DecryptMessage(PCtxtHandle phContext,
 		return SEC_E_INVALID_TOKEN;
 
 	/* Find the proper key and key usage */
-	if (flags & FLAG_ACCEPTOR_SUBKEY)
-		key = context->acceptor_key;
-	else
-		key = context->initiator_key ? context->initiator_key : context->session_key;
-	if (!key)
+	key = get_key(&context->keyset);
+	if (!key || (flags & FLAG_ACCEPTOR_SUBKEY && context->keyset.acceptor_key != key))
 		return SEC_E_INTERNAL_ERROR;
 	usage = context->acceptor ? KG_USAGE_INITIATOR_SEAL : KG_USAGE_ACCEPTOR_SEAL;
 
 	/* Fill in the lengths of the iov array */
 	iov[1].data.length = data_buffer->cbBuffer;
 	iov[2].data.length = 16;
-	if (krb5_c_crypto_length_iov(context->ctx, krb5_k_key_enctype(context->ctx, key), iov,
-	                             ARRAYSIZE(iov)))
+	if (krb5glue_crypto_length_iov(context->ctx, key, iov, ARRAYSIZE(iov)))
```

Comment:
does not return a boolean type, please make this a check against a value.

Reply:
This is how we've done it throughout the code, we're checking for any non-zero value since zero indicates success. If we change it, we should change it everywhere (and while we're at it, log the error in these places too)